cmd/autoscaler/main.go

```diff
@@ -201,8 +164,81 @@ func main() {
 	case <-egCh:
 	case <-stopCh:
 	}
+}
+
+func setupLogger() (*zap.SugaredLogger, zap.AtomicLevel) {
+	loggingConfigMap, err := configmap.Load("/etc/config-logging")
+	if err != nil {
+		log.Fatalf("Error loading logging configuration: %v", err)
+	}
+	loggingConfig, err := logging.NewConfigFromMap(loggingConfigMap)
+	if err != nil {
+		log.Fatalf("Error parsing logging configuration: %v", err)
+	}
+	return logging.NewLoggerFromConfig(loggingConfig, component)
+}
+
+func setupChannels() (<-chan struct{}, chan *autoscaler.StatMessage) {
+	stopCh := signals.SetupSignalHandler()
+	statsCh := make(chan *autoscaler.StatMessage, statsBufferLen)
+	return stopCh, statsCh
+}
+
+func setupClients(logger *zap.SugaredLogger, stopCh <-chan struct{}) (kubernetes.Interface, clientset.Interface, scale.ScalesGetter) {
+	cfg, err := clientcmd.BuildConfigFromFlags(*masterURL, *kubeconfig)
+	if err != nil {
+		logger.Fatalw("Error building kubeconfig", zap.Error(err))
+	}
+
+	kubeClientSet, err := kubernetes.NewForConfig(cfg)
+	if err != nil {
+		logger.Fatalw("Error building kubernetes clientset", zap.Error(err))
+	}
+
+	if err := version.CheckMinimumVersion(kubeClientSet.Discovery()); err != nil {
+		logger.Fatalf("Version check failed: %v", err)
+	}
```

Comment:
Should this check be in the main function itself after `setupClients` was called?

Reply:
I see that this check is extra from creating kubeClientSet.  I tried putting it in main() as you suggested, and it doesn't look as good imo.  It is extra code that you'd need to read without learning about the autoscaler's composition.  And the check can be thought as part of the setup, validating the created client.